woocommerce/google-listings-and-ads
# PMax Assets: Add step 2 to the campaign editing page

### Changes proposed in this Pull Request:

This PR implements parts of 📌 [Add step 2 to the campaign management page](https://github.com/woocommerce/google-listings-and-ads/issues/1787#fe-campaign-step2) in #1787.

- Add step 2 to the campaign editing page and the support of starting from step 2
- Make `AdsCampaign` component support editing a campaign
- Add some editing logic to `AssetGroup` component

It also adds a check to facilitate debugging.

- Add a check for the existence of a context provider to `useAdaptiveFormContext`

### Screenshots:

https://user-images.githubusercontent.com/17420811/205005011-a6c2db43-9da9-4829-bada-a9c739e22755.mp4

### Detailed test instructions:

1. Edit a campaign via the edit button on its most right side.
   - When entering the editing page, it should start with the first step.
   - Check if the stepper works correctly.
   - Update campaigns individually using "Save changes" and "Skip adding assets" buttons.
1. Go back to the Dashboard page.
1. Edit a campaign via the edit button on its "Assets" column.
   - When entering the editing page, it should start with the second step.
   - Update campaigns individually using "Save changes" and "Skip adding assets" buttons.

### Changelog entry
